content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md

```diff
@@ -189,6 +190,20 @@ or `kubeadm upgrade apply`), kubeadm respects the value of `UnversionedKubeletCo
 (during `kubeadm join`, `kubeadm reset`, `kubeadm upgrade ...`), kubeadm attempts to use unversioned ConfigMap name first;
 if that does not succeed, kubeadm falls back to using the legacy (versioned) name for that ConfigMap.
 
+`UpgradeAddonsBeforeControlPlane`
+: This is as a **disabled** feature gate that was introduced for Kubernetes v1.28, in order to allow reactivating a legacy
+and deprecated behavior during cluster upgrade. For kubeadm versions prior to v1.28, kubeadm upgrades cluster addons (including
+CoreDNS and kube-proxy) immediately during `kubeadm upgrade apply`, regardless of whether there are other control plane
+instances that have not been upgraded, which may cause compatibility problems. Since v1.28, kubeadm defaults to a mode that
+always checks whether all the control plane instances have been upgraded before starting to upgrade the addons, this behavior
+is applied to both `kubeadm upgrade apply` and `kubeadm upgrade node`. kubeadm determines whether a control plane instance
+has been upgraded by checking whether the image of the kube-apiserver Pod has been upgraded. You **MUST** perform control plane
+instances upgrade sequentially or at least ensure that the last control plane instance is upgraded after all the other control
+plane instances have been upgraded, and the addons upgrade will be performed after the last control plane instance is upgraded.
+The deprecated `UpgradeAddonsBeforeControlPlane` feature gate gives you a chance to keep the old upgrade behavior. You should not
+need this old behavior; if you do, you should change your cluster or upgrade processes but please do not rely on it, as this
```

Comment:
"please do not rely on it" ...
What does this "it" refer to?

Reply:
"it" means the `UpgradeAddonsBeforeControlPlane` feature gate.

Reply:
"please do not rely on it" has been removed now